lib/sensors/time_of_flight.cpp

```diff
@@ -0,0 +1,180 @@
+    // i2c_->writeByteToRegister(device_address_, 0x01b0, 0x17);
+    // i2c_->writeByteToRegister(device_address_, 0x01ad, 0x00);
+    i2c_->writeByteToRegister(device_address_, 0x00ff, 0x05);
+    // i2c_->writeByteToRegister(device_address_, 0x0100, 0x05);
+    // i2c_->writeByteToRegister(device_address_, 0x0199, 0x05);
+    // i2c_->writeByteToRegister(device_address_, 0x01a6, 0x1b);
+    // i2c_->writeByteToRegister(device_address_, 0x01ac, 0x3e);
+    // i2c_->writeByteToRegister(device_address_, 0x01a7, 0x1f);
+    i2c_->writeByteToRegister(device_address_, 0x0030, 0x00);
+
+    i2c_->writeByteToRegister(device_address_, kSystemFreshOutOfReset, 0);
+
+    // Recommended : Public registers
+    i2c_->writeByteToRegister(device_address_, kSystemModeGpioOne, 0x10);
+    // i2c_->writeByteToRegister(device_address_, kReadoutSamplingPeriod, 0x30);
+    i2c_->writeByteToRegister(device_address_, kSysAlsAnalogueGain, 0x46);
+    i2c_->writeByteToRegister(device_address_, kSysRangeVhvRepeatRate, 0xff);
+    i2c_->writeByteToRegister(device_address_, kSysAlsIntegrationPeriod, 0x63);
+    i2c_->writeByteToRegister(device_address_, kSysRangeVhvRecalibrate, 0x01);
+
+    // Optional: Public registers
+    i2c_->writeByteToRegister(device_address_, kSysRangeIntermeasurementPeriod, 0x09);
+    i2c_->writeByteToRegister(device_address_, kSysAlsIntermeasurementPeriod, 0x31);
+    i2c_->writeByteToRegister(device_address_, kSystemInterruptConfigGpio, 0x24);
+  }
+  return core::Result::kSuccess;
+}
+
+std::optional<std::uint8_t> TimeOfFlight::getRangeSingleShot()
+{
+  if (checkSensorMode(kModeSingleShot) == core::Result::kFailure) { return std::nullopt; }
+  return getRange();
+};
+
+std::optional<std::uint8_t> TimeOfFlight::getRangeContinuous()
+{
+  if (checkSensorMode(kModeContinuous) == core::Result::kFailure) { return std::nullopt; }
+  return getRange();
+};
+
+core::Result TimeOfFlight::checkSensorMode(std::uint8_t mode_value)
+{
+  const auto status = i2c_->readByte(device_address_, kSysRangeStart);
+  if (!status) {
+    logger_.log(core::LogLevel::kFatal, "Failed to check time of flight mode");
+    return core::Result::kFailure;
+  }
+  if (status != mode_value) {
+    const auto start_status
+      = i2c_->writeByteToRegister(device_address_, kSysRangeStart, mode_value);
+    if (start_status == core::Result::kFailure) {
+      logger_.log(core::LogLevel::kFatal, "Failed to set time of flight mode");
+      return core::Result::kFailure;
+    }
+  }
+  return core::Result::kSuccess;
+};
+
+std::optional<std::uint8_t> TimeOfFlight::getRange()
+{
+  // Check the status
+  auto status = i2c_->readByte(device_address_, kResultInterruptStatusGpio);
+  if (!status) {
+    logger_.log(core::LogLevel::kFatal, "Failed to get time of flight interrupt status");
+    return std::nullopt;
+  }
+  auto range_status = *status & 0b111;
+
+  // Wait for new measurement ready status
+  while (range_status != 0x04) {
```

Comment:
this is scary, we need to be able to guarantee that it can't get stuck in this loop